plugins/entando-plugin-jpdigital-exchange/src/main/java/org/entando/entando/aps/system/services/digitalexchange/install/ComponentResource.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2019-Present Entando Inc. (http://www.entando.com) All rights reserved.
+ * 
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ * 
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+package org.entando.entando.aps.system.services.digitalexchange.install;
+
+import com.agiletec.aps.system.exception.ApsSystemException;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import org.entando.entando.aps.system.services.storage.IStorageManager;
+import org.springframework.core.io.AbstractResource;
+
+public class ComponentResource extends AbstractResource {
+
+    private final IStorageManager storageManager;
+    private final String path;
+
+    public ComponentResource(IStorageManager storageManager, String subpath) {
+        this.storageManager = storageManager;
+        this.path = ComponentInstaller.COMPONENTS_DIR + File.separator + subpath;
+    }
+
+    @Override
+    public String getDescription() {
+        return "";
```

Comment:
I'm not sure if I can understand why this description is blank. Is this because the description comes from a different  component object? Maybe add a comment.

Reply:
I left it blank because this method is not used inside the code but the `AbstractResource` class declare it. Now I'm passing the description in the constructor and I added a comment to this class to clarify why it is used.